components/automate-ui/src/app/entities/notification_rules/notification_rule.reducer.ts

```diff
@@ -33,43 +41,113 @@ export function notificationRuleEntityReducer(
 
   switch (action.type) {
     case NotificationRuleActionTypes.GET_ALL:
-      return set(GET_ALL_STATUS, EntityStatus.loading,
-        notificationRuleEntityAdapter.removeAll(state));
+      return set(
+        GET_ALL_STATUS,
+        EntityStatus.loading,
+        notificationRuleEntityAdapter.removeAll(state)
+      ) as NotificationRuleEntityState;
 
     case NotificationRuleActionTypes.GET_ALL_SUCCESS:
       return pipe(
-        set(GET_ALL_STATUS, EntityStatus.loadingSuccess))
-        (notificationRuleEntityAdapter.setAll(action.payload, state));
+        set(GET_ALL_STATUS,
+          EntityStatus.loadingSuccess)
+        )
+        (notificationRuleEntityAdapter.setAll(action.payload, state)
+      ) as NotificationRuleEntityState;
 
     case NotificationRuleActionTypes.GET_ALL_FAILURE:
-      return set(GET_ALL_STATUS, EntityStatus.loadingFailure, state);
+      return set(
+        GET_ALL_STATUS,
+        EntityStatus.loadingFailure,
+        state
+      ) as NotificationRuleEntityState;
```

Comment:
A thing I noticed recently in the code base: we actually do not need type casts on set calls in reducers, only on pipe calls. So you can remove that on every `set` call in this file like this.
```suggestion
      );
```
(Not suggesting to update other files just now, but it is cleaner going forward to do this on files you touch.)